src/controllers/enter.information.controller.ts

```diff
@@ -32,5 +31,28 @@ export const post = async (req: Request, res: Response, next: NextFunction) => {
     return next(e);
   }
 
-  return res.redirect(OBJECTIONS_DOCUMENT_UPLOAD);
+  const changingDetails = retrieveFromObjectionSession(session, CHANGE_ANSWER_KEY);
+  if (changingDetails) {
+    return res.redirect(OBJECTIONS_CHECK_YOUR_ANSWERS);
+  } else {
+    return res.redirect(OBJECTIONS_DOCUMENT_UPLOAD);
+  }
+};
+
+const renderPageWithSessionDataIfPresent = async (req: Request, res: Response, next: NextFunction) => {
+  const objection: Objection = await getObjection(req.session as Session)
+  let existingInformation;
+  if (objection && objection.reason) {
```

Comment:
Should log an error and redirect to error page if objection is missing - same for the session - also we should unit test this scenario.

Reply:
Added some error handling and unit tests to handle if a session or objection is missing